tests/e2e/chain/config.go

```diff
@@ -249,7 +266,7 @@ func initNodes(c *internalChain) error {
 	return nil
 }
 
-func initValidatorConfigs(c *internalChain) error {
+func initValidatorConfigs(c *internalChain, pruning string, pruningKeepRecent string, pruningInterval string, snapshotInterval uint64, snapshotKeepRecent uint32, telemetryEnabled bool, telemetryRetentionTime int64, prometheusEnabled bool) error {
```

Comment:
I think these configs should instead be per-validator. Can we do this in a separate PR? 

I think we should limit the scope of this one to just the upgrade and keep adding on top of it after. Otherwise, we might have to keep iterating on this long PR for some time. Let me know your thoughts